### PR TITLE
Medley of normals and a few other fixes

### PR DESCRIPTION
**Pull request by fabal**
Here are another batch of changes in AL's plugin.
The changes made around normals were aimed at restoring some coherence between the type of mesh (poly vs subdiv) and normals handling. Prior to this there was zero coupling between the which means that exporting subdiv with custom normals was perfectly valid.

### CODE REVIEW

**Comment by fabal** [comment]
The error comes from `AL_USDTransaction` tests in which the python module can't be loaded. It is tricky to debug for me as our build/test setup is different here at Animal and I can't reproduce the error. I did change AL_USDTransaction cmake's config in this PR so there's obviously an issue with the public build. Just let me know what is missing.

Note that I'll be away until Jan 4th.

**Comment by fabal** [comment]
Happy new year! Following up on that one, would you have any insight which would help me solve the remaining error?

File: plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/Transform.cpp [hunk 1]
```diff
@@ -454,12 +461,65 @@ MStatus Transform::validateAndSetValue(
...
+        }
+
+        MTime time;
+        context.getTime(time);
+        MTime theTime = (time - inputTimeValue(dataBlock, m_timeOffset))
+            * inputDoubleValue(dataBlock, m_timeScalar);
+        UsdTimeCode usdTime(theTime.as(MTime::uiUnit()));
+
+        getTransMatrix()->updateToTime(usdTime);
+        MHWRender::MRenderer::setGeometryDrawDirty(thisMObject());
```
Comment (kxl-adsk) [inline plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/Transform.cpp:497]: Notifying renderers about changes in a non-normal context is not allowed. Why this code is necessary?

File: plugin/al/usdtransaction/CMakeLists.txt [hunk 2]
```diff
@@ -1,17 +1,8 @@
...
-  ${CMAKE_CURRENT_SOURCE_DIR}/AL/usd/__init__.py
-  ${CMAKE_CURRENT_SOURCE_DIR}/AL/usd/transaction/__init__.py
-)
-
 # copy to build location for module tests
-foreach(INPUT_FILE ${PY_INIT_FILES})
-  string(REPLACE ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR} OUTPUT_FILE ${INPUT_FILE})
-  execute_process(COMMAND "${CMAKE_COMMAND}" -E copy "${INPUT_FILE}" "${OUTPUT_FILE}")
-endforeach()
+file(COPY AL DESTINATION ${CMAKE_CURRENT_BINARY_DIR} FILES_MATCHING PATTERN "*.py")
```
Comment (kxl-adsk) [inline plugin/al/usdtransaction/CMakeLists.txt:5]: Why does this need to be changed? What are you fixing here?

Comment (fabal) [reply plugin/al/usdtransaction/CMakeLists.txt:5]: It's a clean-up, improving readability.

File: plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/Transform.cpp (hunk 1 above)
Comment (fabal) [reply plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/Transform.cpp:497]: The context of this change was an issue we had with evaluation caching, it dates back to late 2020.
Let me dig into this further.

Comment (fabal) [reply plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/Transform.cpp:497]: I can't answer why it was written that way but I can confirm evaluation caching works properly without it (in fact, it's not needed in post evaluation either).

Comment (fabal) [reply plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/Transform.cpp:497]: Let me update this internally first, then I'll apply the changes here.
In the meantime, any help to solve the remaining failing test would be appreciated.

Comment (fabal) [reply plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/Transform.cpp]: @kxl-adsk , code has been updated.

**Review by kxl-adsk — APPROVED** [review]

**Pull request merged**